fossology/fossology
# feat(copyright): use icu::UnicodeString for IO

## Description

Use `icu::UnicodeString` to support unicode strings. This allows Copyright agent to handle files with unicode content correctly.

As a side-effect, the copyright highlights in PHP UI are currently broken as PHP also needs to be updated to handle unicode files using `mb_` functions. However, doing so breaks highlights of other license agents which also needs to be updated to support unicode.

Thus, this change is broken into 2 stages:
1. First support copyright and other sister agents with unicode supports.
2. Support unicode in license agents and update PHP.

### Changes

1. Use `icu::UnicodeString` to hold any file content instead of `std::string`.
2. Use boost with ICU.
3. Move ICU, LibXslt and OpenMP libraries to CMake way.

## How to test

Run copyright agent on files with unicode content and check the match start and end positions.
